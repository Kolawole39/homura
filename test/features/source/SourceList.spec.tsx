import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { Status } from '../../../app/constants/Status';
import { SourceList } from '../../../app/features/source/SourceList';
import * as sourceSlice from '../../../app/features/source/sourceSlice';

function setup(
  preloadedState: {
    source: sourceSlice.State;
  } = {
    source: {
      list: [],
      activeId: null,
      fetchListStatus: Status.Idle,
      subscribeLink: '',
      subscribeName: '',
      subscribeStep: null,
      subscribeStatus: Status.Idle,
      subscribeError: null,
    },
  }
) {
  const store = configureStore({
    reducer: { source: sourceSlice.sourceReducer },
    preloadedState,
  });

  const wrapper = render(
    <Provider store={store}>
      <Router>
        <SourceList bottom={<div />} />
      </Router>
    </Provider>
  );

  return {
    store,
    wrapper,
  };
}

describe('Source component', () => {
  it('should match exact snapshot', () => {
    const { wrapper } = setup();

    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
