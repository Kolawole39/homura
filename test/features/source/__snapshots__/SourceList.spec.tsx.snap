// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Source component should match exact snapshot 1`] = `
<body>
  <div>
    <div
      class="w-1/6 bg-gray-700 flex flex-col h-screen select-none"
      role="presentation"
    >
      <div
        class="flex-1 thin-scroll relative"
      >
        <div
          class=" leading-8 text-gray-300 flex items-center justify-between cursor-pointer px-3"
          role="button"
          tabindex="0"
        >
          <div
            class="flex items-center overflow-x-hidden"
          >
            <div
              class="h-6 w-6 flex-shrink-0"
            >
              <svg
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 10h16M4 14h16M4 18h16"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="text-xs ml-2 truncate"
            >
              All Items
            </div>
          </div>
        </div>
        <div
          class=" leading-8 text-gray-300 flex items-center justify-between cursor-pointer px-3"
          role="button"
          tabindex="0"
        >
          <div
            class="flex items-center overflow-x-hidden"
          >
            <div
              class="h-6 w-6 flex-shrink-0"
            >
              <svg
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 8h14M5 8a2 2 0 110-4h14a2 2 0 110 4M5 8v10a2 2 0 002 2h10a2 2 0 002-2V8m-9 4h4"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="text-xs ml-2 truncate"
            >
              Archive
            </div>
          </div>
        </div>
        <div
          class="pt-10"
        />
      </div>
      <div />
    </div>
  </div>
</body>
`;
